phoenix_liveview/mix_install_deps_backport.exs

```diff
@@ -0,0 +1,39 @@
+defmodule InstallFolderTemporaryBackport do
```

Comment:
This code is necessary to mimic the computation made by `Mix.install/2` to create the install folders, from where the app needs to find the `package.json` of each JS dependency.

Ideally a bit of refactoring of `Mix.install/2` itself would help expose the correct paths to the app, and remove the need for this.

Reply:
Do you have some concrete changes to Mix.install in mind?

One idea is the following. Mix.install can already be configured via the `MIX_INSTALL_DIR`. Maybe after Mix installing, we'd set the env variable?

Reply:
I will try to explain what is problematic with setting `MIX_INSTALL_DIR` only.

If I run `MIX_INSTALL_DIR=some-folder elixir phoenix_liveview.exs`, the install folder layout will be:

```
❯ tree some-folder -L 3
some-folder
└── elixir-1.12.2-erts-12.0.3
    └── 3592bd731a480b3e42d3e6048caaf789
        ├── _build
        ├── deps
        └── mix.lock
```

This means that while running `MIX_INSTALL_DIR=some-folder elixir phoenix_liveview.exs`,  the running program needs to figure out where is the `deps` folder, and so recompute the whole `elixir-1.12.2-erts-12.0.3/3592bd731a480b3e42d3e6048caaf789`.

Once the caller knows the folder, they can configure whatever they need to build the assets. Here if I hardcode what is necessary with the `tree` output above, it is:

```elixir
Application.put_env(:esbuild, :default,
  args: ~w(app.js --bundle --target=es2016 --outdir=../priv/static/assets),
  cd: Path.expand("assets", __DIR__),
  env: %{
    "NODE_PATH" =>
      Path.expand("some-folder/elixir-1.12.2-erts-12.0.3/3592bd731a480b3e42d3e6048caaf789/deps")
  }
```

To make it easy, it would be necessary to expose (instead of `MIX_INSTALL_DIR`) the `dir` variable to the application, as computed here in Mix:

https://github.com/elixir-lang/elixir/blob/3a7a598c255fdc1e719bc410f77071f7cdf9dd95/lib/mix/lib/mix.ex#L626-L628

(copy-pasted for ease of read:)

```elixir
  id = deps |> :erlang.term_to_binary() |> :erlang.md5() |> Base.encode16(case: :lower)
    version = "elixir-#{System.version()}-erts-#{:erlang.system_info(:version)}"
    dir = Path.join([installs_root, version, id])
```

So to summarise, it would be useful, for anyone using assets coming from `deps` (including phoenix & liveview), to programmatically expose the exact subfolder of installation.

@wojtekmach hope this is clearer, let me know if you think this would be doable!

Reply:
This is now irrelevant, because I have applied @josevalim suggestion in 708e1de6b4710fb8ed55b9eefca71d01ba7cef36 (`Path.expand(Application.app_dir(:phoenix, "../../../../deps"))`) to avoid all this code.